lib/Sema/TypeCheckType.cpp

```diff
@@ -1047,22 +1049,9 @@ static Type SelfAllowedBySE0068(TypeResolution resolution,
       (nominal = nominalDC->getSelfNominalTypeDecl())) {
     assert(!isa<ProtocolDecl>(nominal) && "Cannot be a protocol");
 
-    bool insideClass = nominalDC->getSelfClassDecl() != nullptr;
-    AbstractFunctionDecl *methodDecl = dc->getInnermostMethodContext();
-    bool declaringMethod = methodDecl &&
-      methodDecl->getDeclContext() == dc->getParentForLookup();
-    bool isMutablePropertyOrSubscriptOfClass = insideClass &&
-      (options.is(TypeResolverContext::PatternBindingDecl) ||
-       options.is(TypeResolverContext::FunctionResult));
-    bool isTypeAliasInClass = insideClass &&
-      options.is(TypeResolverContext::TypeAliasDecl);
-
-    if (((!insideClass || !declaringMethod) &&
-         !isMutablePropertyOrSubscriptOfClass && !isTypeAliasInClass &&
-         !options.is(TypeResolverContext::GenericRequirement)) ||
-        options.is(TypeResolverContext::ExplicitCastExpr)) {
+    if (!options.is(TypeResolverContext::GenericRequirement)) {
```

Comment:
Much cleaner. And you can diagnose Self in generic requirements out-of-band too.

Reply:
Can you give me a hint where that would be?